jscomp/syntax/src/res_core.ml

```diff
@@ -2617,10 +2617,11 @@ and parseJsxOpeningOrSelfClosingElement ~startPos p =
     | GreaterThan -> (
       (* <foo a=b> bar </foo> *)
       let childrenStartPos = p.Parser.startPos in
-      Scanner.setJsxMode p.scanner;
       Parser.next p;
       let spread, children = parseJsxChildren p in
       let childrenEndPos = p.Parser.startPos in
+      Scanner.popMode p.scanner Jsx;
+      Scanner.setJsxMode p.scanner;
```

Comment:
by using `popMode` before `setJsxMode` we make sure we don't add too many times the Jsx mode to the scanner mode list.

I also made sure we were calling `setJsxMode` at the right place, which was not really the case before and led to invalid results (allowing hyphens in contexts that should not have been in Jsx mode).